kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -189,40 +240,32 @@ def after_catalog_created(self, catalog):
         )
 
 
-def _is_known_ci_env(known_ci_env_var_keys=KNOWN_CI_ENV_VAR_KEYS):
+def _is_known_ci_env(known_ci_env_var_keys: set[str]):
     # Most CI tools will set the CI environment variable to true
     if os.getenv("CI") == "true":
         return True
     # Not all CI tools follow this convention, we can check through those that don't
     return any(os.getenv(key) for key in known_ci_env_var_keys)
 
 
-def _get_project_properties(user_uuid: str, project_path: Path) -> dict:
-    hashed_package_name = _hash(str(PACKAGE_NAME)) if PACKAGE_NAME else "undefined"
+def _get_project_properties(user_uuid: str, pyproject_path: Path) -> dict:
+    project_uuid = _get_or_create_project_uuid(pyproject_path)
+    package_name = PACKAGE_NAME or UNDEFINED_PACKAGE_NAME
+    hashed_project_uuid = (
+        _hash(f"{project_uuid}{package_name}") if project_uuid is not None else None
```

Comment:
Reading this logic again. The only case in which `_get_or_create_project_uuid(pyproject_path)` will return `None` is when the `pyproject.toml` file cannot be found, am I right?

Reply:
Yes, in all other cases, we either read it from there or generate and append it like so:
```
[tool.kedro_telemetry]
project_id = "..."
```